rwengine/src/script/ScriptMachine.cpp

```diff
@@ -148,12 +148,18 @@ void ScriptMachine::executeThread(SCMThread &t, int msPassed)
 #if RW_SCRIPT_DEBUG
 		if (strcmp(t.name, "EIGHT") == 0)
 		{
-			printf("% 8s  %04x %04x % 25s", t.name, t.programCounter, opcode, code.signature.c_str());
+			printf("% 8s %01x %04x %04x % 25s", t.name, t.conditionResult, t.programCounter, opcode, code.signature.c_str());
 			for (auto& a : sca.getParameters())
 			{
-				printf(" %08x", a.integerValue());
+				if (a.type == SCMType::TString) {
+					printf(" %01x:% 8s", a.type, a.string);
+				}
+				else {
+					printf(" %01x:%08x", a.type, a.integerValue());
+				}
 			}
 			printf("\n");
+			fflush(0);
```

Comment:
a) This seems unnecessary since stdout/stderr are usually line-buffered.

b) Even if it was, you should at least write `fflush(stdout);` or whatever (be explicit)